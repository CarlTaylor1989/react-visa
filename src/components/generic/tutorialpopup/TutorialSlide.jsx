import React from 'react';
import PropTypes from 'prop-types';

import './TutorialSlide.scss';

const TutorialSlide = ({ imageSrc, title, text }) => {
  const imgUrl = require(`../../../resources/tutorials/${imageSrc}`); // eslint-disable-line
  return (
    <div className="tutorialSlide">
      <div>
        <img
          className="slideImg"
          src={imgUrl}
          alt=""
        />
      </div>
      <div className="contentArea">
        {title && (
          <div
            className="title"
            dangerouslySetInnerHTML={{ __html: title }} // eslint-disable-line react/no-danger
          />
        )}
        {text && (
          <div
            className="text"
            dangerouslySetInnerHTML={{ __html: text }} // eslint-disable-line react/no-danger
          />
        )}
      </div>
    </div>
  );
};

TutorialSlide.propTypes = {
  imageSrc: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default TutorialSlide;
