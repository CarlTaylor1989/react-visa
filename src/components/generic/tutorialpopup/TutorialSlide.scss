@import '../../../styles/mixins';
@import '../../../styles/variables';

$persona-icon: '../../../styles/icons/tutorials/tutorial_persona_icon.png';
$persona-icon-width: 100px;
$persona-icon-height: 100px;

.tutorialSlide {
  @include align-items(flex-start);
  display: flex;
  justify-content: space-between;

  .contentArea {
    background-image: url($persona-icon);
    background-position: top center;
    background-repeat: no-repeat;
    background-size: $persona-icon-width $persona-icon-height;
    margin-left: 30px;
    padding-top: 120px;

    .title {
      color: $colour-gold;
      font-size: 30px;
      margin-bottom: 30px;
    }

    .text {
      color: $colour-gold;
      font-size: 22px;
    }
  }
}
