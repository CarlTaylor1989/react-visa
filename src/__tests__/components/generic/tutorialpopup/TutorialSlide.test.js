/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import TutorialSlide from '../../../../components/generic/tutorialpopup/TutorialSlide';

let mockProps;
beforeEach(() => {
  mockProps = {
    imageSrc: 'foo.png',
    text: '',
    title: ''
  };
});

test('renders the TutorialSlide component with just a title', () => {
  const wrapper = shallow(<TutorialSlide {...mockProps} title="irrelevant" />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the TutorialSlide component with just some text', () => {
  const wrapper = shallow(<TutorialSlide {...mockProps} text="irrelevant" />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
