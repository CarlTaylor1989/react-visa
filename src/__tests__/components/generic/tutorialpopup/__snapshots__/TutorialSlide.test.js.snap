// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TutorialSlide component with just a title 1`] = `
<div
  className="tutorialSlide"
>
  <div>
    <img
      alt=""
      className="slideImg"
      src=""
    />
  </div>
  <div
    className="contentArea"
  >
    <div
      className="title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "irrelevant",
        }
      }
    />
  </div>
</div>
`;

exports[`renders the TutorialSlide component with just some text 1`] = `
<div
  className="tutorialSlide"
>
  <div>
    <img
      alt=""
      className="slideImg"
      src=""
    />
  </div>
  <div
    className="contentArea"
  >
    <div
      className="text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "irrelevant",
        }
      }
    />
  </div>
</div>
`;
